README.md

```diff
@@ -154,6 +154,44 @@ One the values that a given visitor will be assigned for a split, e.g. `true` or
 ### Weighting
 Variants are assigned pseudo-randomly to visitors based on their visitor IDs and the weightings for the variants.  Weightings describe the probability of a visitor being assigned to a given variant in integer percentages.  All the variant weightings for a given split must sum to 100, though variants may have a weighting of 0.
 
+### Experiment
+
+Experiments are the standard flavor of splits in TestTrack. They are
+intended to be used for A/B testing, and the TestTrack server records
+visitors' experienced variants so that those visitors will continue to
+experience the same variant regardless of subsequent changes to the
+weightings of those variants via the admin interface.
+
+Storing the variant a visitor experienced for an experiment also allows
+TestTrack to provide a consistent UX to a customer who experienced a
+new-to-them experiment before logging in on a new device, only to be
+recognized as an existing visitor upon sign-in.  TestTrack will merge
+all variant assignments from the anonymous visitor into the
+authenticated visitor at sign-in as long as the authenticated visitor
+doesn't have conflicting assignments.  In that case, the authenticated
+visitor's previous assignments win.
+
+### Feature Gate
+
+As of TestTrack version 1.2, splits with names ending in the `_enabled`
+suffix will be treated as feature gates. Feature gates differ from
+experiments in that they are not intended to be used for analysis, and
+therefore it is not important that the user remain in the same variant
+throughout the entire split lifecycle. Feature gates are meant to be
+slow-rolled (incrementally increasing the percentage of customers
+experiencing the new feature), released en masse, or instantly rolled
+back.
+
+To facilitate these smooth transitions and rapid toggles, the TestTrack
+server will not record variant assignments when a visitor experiences a
+split. This means that every time a visitor experiences a split, they
+will be deterministically (pseudorandomly) assigned to a variant based
+on their visitor ID and the name of the split. This will provide the
+customer with a stable variant given a constant split weighting, but
+probablistically increase the size of the population experiencing the
+the `true` variant as the split weightings are increased via the admin
+panel, giving an admin full control over the feature's release.
```

Comment:
won't we only store the assignment if a customer is explicitly assigned via the admin panel? i.e. if we just increase the `true` weighting on incremental rollout, couldn't a customer randomly experience the `true` version and then come back and randomly experience the `false` version? so we _have_ to incrementally release by explicit assignment of customer ids? or am i missing something?

Reply:
The algo works something like this:

truncate_to_integer_and_scale_to_number_between_0_and_99(sha1(visitor_id + split_name))

we then sort the variants alphabetically and then assign them ranges making up the numbers between 0 and 99. You experience the variant that your number buckets to.

So as we ratchet up or down, the same users for a given split will be rolled out to and rolled back from in deterministic, pseudorandom order per split.

That said, I probably shouldn't use the word population in this definition in a way that differs from the population in the admin - none of these visitors will show up in the population in the admin, because they won't have any concrete assignments.

Reply:
(this would work horribly if you had more than two variants in a gate as the middle variant would slide through the population, but what kind of gate has more than two sides? this is why we wanna formalize the API for making gates.)